pages/user/[uid].js

```diff
@@ -61,40 +61,69 @@ let tempRecipes = [{
 	]
 }] //TODO: remove once db val used
 
+
+function filterToUserData(items, idString){
+	const ids = new Set(JSON.parse(idString))
+	return items.filter((item) => ids.has(item.id))
+}
+
 export async function getServerSideProps(context) 
 {
-	const uid = context.uid
-	const ingredients = [
-		{ title: 'Milk'}, { title: 'Bread' },  { title: 'Cheddar' },  { title: 'Coffee' },  { title: 'Eggs' }
-	]
-	const myIngredients = [
-		{ title: 'Milk'}
-	]
-	const restrictions = [
-		{ title: 'Gluten Free' }, { title: 'Vegetarian' }, { title: 'Vegan' }, { title: 'Peanut Allergy' }
-	]
-	const myRestrictions = []
-	const equipment = [
-		{ title: 'Stove' }, { title: 'Microwave' }, { title: 'Beater' }, { title: 'Cutting Board' }, { title: 'Freezer' }, { title: 'Pressure Cooker' }
-	]
-	const myEquipment = [
-		{ title: 'Stove' }
-	]
+	const userCont = require("../../backend/controllers/user_controller")
+	const equipmentCont = require("../../backend/controllers/equipment_controller")
+	const ingredientCont = require("../../backend/controllers/ingredient_controller")
+	const recipeCont = require("../../backend/controllers/recipe_controller")
+	//const restriction = require("../../backend/controllers/restriction_controller")
+	const uid = context.params.uid
+	const userProm = userCont.get(uid)
+	const ingredientsProm = ingredientCont.getAll()
+	//const restrictions = restriction.getAll()
+	//const recipesProm = recipeCont.getAll()
+	const equipmentProm = equipmentCont.getAll()
+
+	//var [user, ingredients, equipment, recipes] = await Promise.all([userProm, ingredientsProm, equipmentProm, recipesProm])
+	var [user, ingredients, equipment] = await Promise.all([userProm, ingredientsProm, equipmentProm])
+	ingredients = ingredients.map(function(ingredient) {
+		return {id: ingredient.dataValues.id, title: ingredient.dataValues.name}
+	})
+	equipment = equipment.map(function(equipment) {
+		return {id: equipment.dataValues.id, title: equipment.dataValues.name}
+	})
 
+	console.log("fetched user")
+	console.log(user)
+
+	const myIngredients = filterToUserData(ingredients, user.dataValues.ingredients)
+	const myEquipment = filterToUserData(equipment, user.dataValues.equipment)
+	//const myRecipes = filterToUserData(recipes, user.dataValues.recipes)
+	
+	//const myRestrictions = filterToUserData(restrictions, user.restrictions)
+	//const myRecipes = filterToUserData(recipes, user.recipes)
+
+	//const ingredients = []
+	//const myIngredients = []
+	//const equipment = []
+	//const myEquipment = []
```

Comment:
Remove commented out stubs

Reply:
leaving in the top comments, since that kind of method call will be needed for when restriction and recipes data fetching is implemented